lib/reporting/reports/products_and_inventory/all_products.rb

```diff
@@ -4,9 +4,43 @@ module Reporting
   module Reports
     module ProductsAndInventory
       class AllProducts < Base
+        def default_params
+          { fields_to_hide: [:on_demand, :on_hand] }
+        end
+
+        def columns
+          super.merge(
+            {
+              on_demand: proc{ |variant| variant.stock_items.first.backorderable },
```

Comment:
I'm still trying to understand the concepts "backorderable" and "on demand" (unfortunately they don't really seem to be covered in the [user guid](https://guide.openfoodnetwork.org/)).
Is there a reason to check `stock_items.first.backorderable` instead of `product.on_demand` here? (and also in `on_hand` below)

Reply:
I think this is because `on_demand` is an attribute related to the variant not to the product.
![image](https://user-images.githubusercontent.com/16113997/205005322-374e8020-e327-4d13-8c56-fab7faa62d0a.png)
And if the product has more than one variant, the `Product#on_demand` will raise an exception. 

https://github.com/openfoodfoundation/openfoodnetwork/blob/master/app/models/concerns/product_stock.rb#L8
```ruby
  def on_demand
    raise 'Cannot determine product on_demand value of product with multiple variants' if variants.size > 1

    variants.first.on_demand
  end
```

Reply:
@dacook 
I agree. I didn't notice that method.
We can do the same thing for `on_hand` which is also implemented on the `variant` model.

Reply:
@dacook done.